### PR TITLE
Related Products: hide unusable options from Inspector Controls

### PR DESCRIPTION
**Pull request by kmanijak**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

@jarekmorawski brought to my attention Related Products offer some controls that are unusable:
- Advanced Filters (valid in products (Beta) but not here)
- Upgrade Notice - Product Collection doesn't support Related Products case and clicking the upgrade does nothing.

This PR hides both sections.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to Editor -> Single Product template
2. Add Related Products block if there isn't one
3. Focus on "Related Products Controls"
4. Expected: there's no "Advanced Filters" or "Upgrade Notice" sections in Inspector Controls (check screenshot below)

Before | After
-- | --
![Screen Shot 2024-05-27 at 14 56 41 PM]
image | image

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

**Review by thealexandrelara — CHANGES_REQUESTED** [review]
Thank you for working on this! Everything is working as expected. 

Considering the upgrade notice and the advanced filters require specific conditions to appear, it would be great if you could include an E2E test covering them 🙌 .

**Comment by kmanijak** [comment]
> Considering the upgrade notice and the advanced filters require specific conditions to appear, it would be great if you could include an E2E test covering them 🙌 .

Added E2E test! @thealexandrelara can I ask you for re-review?

File: plugins/woocommerce-blocks/tests/e2e/tests/related-products/related-products.block_theme.spec.ts [hunk 1]
```diff
@@ -63,4 +63,54 @@ test.describe( `${ blockData.name } Block`, () => {
 			await editor.getBlockByName( blockData.slug )
 		).toBeVisible();
 	} );
+
+	test( "doesn't display additional Products (Beta) options", async ( {
+		page,
+		admin,
+		editor,
+		editorUtils,
```
Comment (thealexandrelara) [inline plugins/woocommerce-blocks/tests/e2e/tests/related-products/related-products.block_theme.spec.ts]: It seems that `editorUtils` does not exist anymore so it's causing an error in the E2E tests. You should use `editor` instead.

```suggestion
```

File: plugins/woocommerce-blocks/tests/e2e/tests/related-products/related-products.block_theme.spec.ts [hunk 2]
```diff
@@ -63,4 +63,54 @@ test.describe( `${ blockData.name } Block`, () => {
...
+	} ) => {
+		await admin.visitSiteEditor( {
+			postId: `woocommerce/woocommerce//single-product`,
+			postType: 'wp_template',
+		} );
+		await editorUtils.enterEditMode();
+		const relatedProducts = await editorUtils.getBlockByName(
+			blockData.slug
+		);
```
Comment (thealexandrelara) [inline plugins/woocommerce-blocks/tests/e2e/tests/related-products/related-products.block_theme.spec.ts]: You can use `editor` here:

```suggestion
		const relatedProducts = await editor.getBlockByName(
			blockData.slug
		);
```

File: plugins/woocommerce-blocks/tests/e2e/tests/related-products/related-products.block_theme.spec.ts [hunk 3]
```diff
@@ -63,4 +63,54 @@ test.describe( `${ blockData.name } Block`, () => {
...
+		// To confirm test is correct, we verify if Products (Beta)
+		// have these options available using the same locators.
+		await editor.setContent( '' );
+		const querySlug = 'core/query';
+		const productsBetaSlug = 'woocommerce/product-query';
+		await editor.insertBlock( {
+			name: querySlug,
+			attributes: { namespace: productsBetaSlug },
+		} );
+		const productsBeta = await editorUtils.getBlockByName( querySlug );
```
Comment (thealexandrelara) [inline plugins/woocommerce-blocks/tests/e2e/tests/related-products/related-products.block_theme.spec.ts]: This should be replaced by `editor`:

```suggestion
		const productsBeta = await editor.getBlockByName( querySlug );
```

File: plugins/woocommerce-blocks/tests/e2e/tests/related-products/related-products.block_theme.spec.ts [hunk 4]
```diff
@@ -63,4 +63,54 @@ test.describe( `${ blockData.name } Block`, () => {
...
+		page,
+		admin,
+		editor,
+		editorUtils,
+	} ) => {
+		await admin.visitSiteEditor( {
+			postId: `woocommerce/woocommerce//single-product`,
+			postType: 'wp_template',
+		} );
+		await editorUtils.enterEditMode();
```
Comment (thealexandrelara) [inline plugins/woocommerce-blocks/tests/e2e/tests/related-products/related-products.block_theme.spec.ts]: This should be `editor`:

```suggestion
		await editor.enterEditMode();
```

**Review by thealexandrelara — DISMISSED** [review]
Hey @kmanijak, thanks for adding the E2E tests . I noticed that the E2E tests are failing to run because you're using `editorUtils`, which seems no longer available. So, I believe you should be using `editor` instead, I commented in the code on the places you can replace it.

**Comment by kmanijak** [comment]
> Hey @kmanijak, thanks for adding the E2E tests . I noticed that the E2E tests are failing to run because you're using editorUtils, which seems no longer available. So, I believe you should be using editor instead, I commented in the code on the places you can replace it.

Oh, I merged `trunk` and didn't realize there was such change! Thanks for pointing it out and sorry I tagged you for re-review with the failing tests 😅 Now, can I ask for (hopefully) final re-review @thealexandrelara ?

File: plugins/woocommerce-blocks/tests/e2e/tests/single-product-template/single-product-template-compatibility-layer.spec.ts [hunk 5]
```diff
@@ -83,24 +83,22 @@ const singleOccurranceScenarios: Scenario[] = [
...
 	} );
 
-	test.describe( 'Product Archive with Product Collection block', () => {
-		for ( const scenario of singleOccurranceScenarios ) {
-			test( `${ scenario.title } is attached to the page`, async ( {
-				page,
-			} ) => {
-				await page.goto( '/product/hoodie/' );
-				const hooks = page.getByTestId( scenario.dataTestId );
+	for ( const scenario of singleOccurranceScenarios ) {
```
Comment (WunderBart) [inline plugins/woocommerce-blocks/tests/e2e/tests/single-product-template/single-product-template-compatibility-layer.spec.ts]: nit: 
```suggestion
	for ( const scenario of singleOccurrenceScenarios ) {
```

**Review by imanish003 — APPROVED** [review]
I tested the changes & verified that Advanced filters and Upgrade Notice aren't visible in the Related Products block. Thanks for quickly fixing this 🚀

**Pull request merged**